docs/i18n.en-US.md

```diff
@@ -0,0 +1,216 @@
+...
+constructor(private nzI18nService:NzI18nService) {
+}
+
+switchLanguage() {
+  this.nzI18nService.setLocale(en_US);
+}
+```
+
+### @delon
+
+@delon internationalization defaults to Chinese version, for example the default is English version:
+
+```ts
+import { DELON_LOCALE, en_US } from '@delon/theme';
+@NgModule({
+  ...
+  providers   : [ { provide: DELON_LOCALE, useValue: en_US } ]
+})
+export class AppModule { }
+```
+
+Of course, you can also use runtime changes:
+
+```ts
+import { en_US, DelonLocaleService } from '@delon/theme';
+...
+constructor(private delonLocaleService: DelonLocaleService) {
+}
+
+switchLanguage() {
+  this.delonLocaleService.setLocale(en_US);
+}
+```
+
+## ALAIN_I18N_TOKEN
+
+`@delon/*` class library has many data interface properties with the _i18n_ typeface (for example: `page-header`, `st` column description, `Menu` menu data, etc.) when you want the data for these components. When the interface can dynamically switch automatically according to the Key value in the current language, you also need to define a self-implementation service interface for `ALAIN_I18N_TOKEN` (for example: [I18NService](https://github.com/ng-alain/ng-alain/blob) /master/src/app/core/i18n/i18n.service.ts)) and register under the root module.
+
+```ts
+import { ALAIN_I18N_TOKEN } from '@delon/theme';
+import { I18NService } from '@core/i18n/i18n.service';
+
+@NgModule({
+  ...
+  providers: [
+    { provide: ALAIN_I18N_TOKEN, useClass: I18NService, multi: false }
+  ]
+})
+export class AppModule {}
+```
+
+### i18n pipe
+
+In order not to be named by the third-party pipes, the scaffolding contains a `i18n` pipe, which is equivalent to calling the `fanAIN` method of `ALAIN_I18N_TOKEN` directly.
+
+That differs from `@ngx-translate`'s `|translate`, which listens for language changes and updates automatically. `| i18n` will not listen to language change notifications, so there will be better performance. When you explicitly re-render the Angular project after switching languages, `| i18n` will be more suitable.
+
+## How to add
+
+When creating scaffolding [from command line](/cli/add) `ng add ng-alain`, it is allowed to specify `--i18n` to indicate whether the internationalized sample code is included (in the `@ngx-translate/core` mode).
+
+## How to delete
+
+The sample code covers the following:
+
+- `src/app/core/i18n` directory
+- `app.module.ts` for `TranslateModule`
+- Replace the pipe of i18n that may appear in the default layout using `| translate` or `| i18n`
+- Remove `@ngx-translate/core`, `@ngx-translate/http-loader` package
+
+## Default language
+
+Regardless of whether internationalization is required or not, since the default languages ​​of class libraries such as `Angular`, `ng-zorro-antd`, `@delon/*` are different, it is also necessary to ensure that the default language of these libraries is ** the same type* *. A simple example approach to understand the current language situation for each type of library:
```

Comment:
```suggestion
Regardless of whether internationalization is required or not, since the default languages ​​of class libraries such as `Angular`, `ng-zorro-antd`, `@delon/*` are different, it is also necessary to ensure that the default language of these libraries is **the same type**. A simple example approach to understand the current language situation for each type of library:
```